src/components/TodoMain/TodoItem.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { ErrorType } from '../../types/ErrorType';
+
+type Props = {
+  todo: Todo,
+  selectedTodoIds: number[],
+  onDelete: (todoId: number) => void,
+  updateTodo: (updatedTodo: Todo) => void,
+  onError: (error: ErrorType) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  selectedTodoIds,
+  onDelete,
+  updateTodo,
+  onError,
+}) => {
+  const isLoading = selectedTodoIds.includes(todo.id) || todo.id === 0;
+  const [isBeingEdited, setIsBeingEdited] = useState(false);
+  const [titleOnEdit, setTitleOnEdit] = useState(todo.title);
+
+  const titleInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    titleInput.current?.focus();
+  }, [isBeingEdited]);
+
+  const handleTodoStatusChange = async (todoToUpdate: Todo) => {
+    const updatedTodo: Todo = {
+      ...todoToUpdate,
+      completed: !todoToUpdate.completed,
+    };
+
+    updateTodo(updatedTodo);
+  };
+
+  const onEscKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setIsBeingEdited(false);
+      setTitleOnEdit(todo.title);
+    }
+  };
+
+  const onSubmitTitle = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (titleOnEdit === todo.title) {
+      setIsBeingEdited(false);
+
+      return;
+    }
+
+    const updatedTodo: Todo = {
+      ...todo,
+      title: titleOnEdit.trim(),
+    };
+
+    if (!updatedTodo.title) {
+      onDelete(todo.id);
+
+      return;
+    }
+
+    try {
+      updateTodo(updatedTodo);
+      setIsBeingEdited(false);
+    } catch {
+      onError(ErrorType.UnableToUpdate);
+      setIsBeingEdited(true);
+    }
```

Comment:
- If you handle errors in the `updateTodo` function you don't need to do the same here.
- `setIsBeingEdited(false);` will be executed before than async function fihish
- `setIsBeingEdited(false)` you do it in both cases, operations that you do anyway should be placed in finally block.

So looks like it will be enough:
```javascript
await updateTodo(updatedTodo);
setIsBeingEdited(false);
```